### ISSUE TITLE
Incorrect warning for SC2051 with eval printf.

### ISSUE THREAD

**Issue by xzovy** [koalaman/shellcheck#583]
Warns about the following:
```shell
length=5
eval printf "%.0s-" {1..$length}
```
even though that operation works perfectly.

**Comment by arth1** [comment]
@xzovy Not if length is less than 1, it doesn't...
Or, try length=9999999999999999 or so, and you'll get an xmalloc error and the eval exits the shell, which is not good.

But yeah, bash does not support variables in brace range expansions, but the eval certainly does.

**Comment by xzovy** [comment]
Neither of those issues were caused by the use of a variable. If you remove the variable from the command and run it with a negative number or a massive number you get the same result as if you were storing that number in a variable. 

Also eval is a bash builtin, so what does it mean to say that bash doesn't support it but eval does? 

Lastly, I can make shellcheck stop complaining about the variable inside the brace expansion simply by double quoting the brace expansion. While one might want to double quote the expansion, it doesn't make any sense for that to resolve SC2051.

I feel this is a bug in shellcheck because even if one shouldn't be using the command above, though I have found it useful, the warning given by shellcheck does not apply.

**Comment by deryni** [comment]
This is a duplicate of the recently fixed #580 I think.

**Comment by xzovy** [comment]
Correct me if I am wrong, but it appears that fix only applies to instances where echo is used, not printf. If this is incorrect, when do you think this fix will make it to shellcheck.net?

**Comment by koalaman** [comment]
It was rolled back due to reasons. Another fix is incoming.

**Comment by xzovy** [comment]
Wonderful. I look forward to it. Until then, I will just put double quotes around my brace expansions since there is no real downside (as far as I can tell).

**Comment by koalaman** [comment]
Fixed in 5f1175fb. This replaces it with one suggesting that you double quote, since relying on the shell to silently fail and pass it on literally is a strange thing to do.

**Comment by deryni** [comment]
Expecting the shell to fail and pass on the literal is a "bug" that bit the bash-completion project for a while. Turning on `nullglob` would break many completions. =)

Though to be fair to the bash-completion people they probably didn't know that's what they were counting on.

See http://stackoverflow.com/a/29909046/258523 for where I looked into it and found where they fixed it if you are curious.

**Comment by koalaman** [comment]
Haha, glorious.

There should probably be some kind of check for that.
